docs: add map information for clone images (#56160)
ceph/ceph: doc/rbd/rbd-encryption.rst

(plaintext) image since an unformatted image does not have a header at
all.

> To map a formatted clone of an encrypted parent image, provide the
> passphrase file of the clone and its encrypted parent image. The order in
> which the passphrase files should be provided is based on the image
> hierarchy, starting with that of the clone image, then its parent and so on.

Reviewer: More precise language:
```suggestion
To map a formatted clone, provide encryption formats and passphrases
for the clone itself and all of its explicitly formatted parent images.
The order in which `encryption-format` and `encryption-passphrase-file`
options should be provided is based by the image hierarchy: start with
that of the cloned image, then its parent and so on.
```

Reviewer 2: This has been committed as it stands here with the exception of "s/is based by/is based on/".

Reviewer: I actually edited that before your comment ;)

> .. prompt:: bash $

Reviewer: This command requires privileges:
```suggestion
.. prompt:: bash #
```

    rbd device map -t nbd -o encryption-passphrase-file=clone-passphrase.bin -o encryption-passphrase-file=passphrase.bin mypool/myclone

> 

Reviewer 3: LGTM. Is it worth the 2 additional blank spaces ?

> 

Reviewer: Are that many blank lines needed for better rendering?  If not, let's leave just one.

(plaintext) image since an unformatted image does not have a header at
all.

To map a formatted clone, provide encryption formats and passphrases
for the clone itself and all of its explicitly formatted parent images.
The order in which ``encryption-format`` and ``encryption-passphrase-file``
options should be provided is based on the image hierarchy: start with
that of the cloned image, then its parent and so on.

Here is an example of a command that maps a formatted clone:

.. prompt:: bash #

>    rbd device map -t nbd -o encryption-passphrase-file=clone-passphrase.bin -o encryption-passphrase-file=passphrase.bin mypool/myclone

Reviewer 3: ```suggestion
   rbd device map -t nbd -o encryption-passphrase-file=clone-passphrase.bin,encryption-passphrase-file=passphrase.bin mypool/myclone
```

Reviewer 3: In addition to above can we also have another example like: (or if we want to keep one eg, then keep the most detailed one)

```
In case if parent image and clone are formatted with different encryption formats:

rbd device map -t nbd -o  encryption-format=luks2,encryption-passphrase-file=clone-passphrase.bin,encryption-format=luks1,encryption-passphrase-file=passphrase.bin mypool/myclone
```

Reviewer 3: @zdover23 if Ilya doesn't show-up soon, please feel free to go ahead and merge, this is not the end of the world :-)

Thank you!

Reviewer 2: @pkalever: Roger that. I'll merge as soon as it's possible, but it might not be for several hours. In any case, this issue is in my list of things to finish by the end of the workday. It will be done.

Reviewer: > In addition to above can we also have another example like: (or if we want to keep one eg, then keep the most detailed one)

I got rid of `encryption-format` option in these examples on purpose in commit a27ee2bdf87ca1419418f3b6c8c00cdf335d7b43.  It's optional and these commands are long/verbose enough, no need to pile up ;)

Switching to a single `-o` is fine with me.  If it seems clearer to folks, let's incorporate that.

Reviewer 2: See https://github.com/ceph/ceph/pull/56188.
